config/suit/templates/nrf54h20/default/v1/root_with_binary_nordic_top.yaml.jinja2

```diff
@@ -185,77 +217,180 @@ SUIT_Envelope_Tagged:
+{%- endif %}
+{%- if radio is defined and application is defined %}
+      - - suit-directive-set-component-index: 0
+        - suit-directive-override-parameters:
+            suit-parameter-uri: '#{{ radio['name'] }}'
+            suit-parameter-image-digest:
+              suit-digest-algorithm-id: cose-alg-sha-256
+              suit-digest-bytes:
+                envelope: {{ artifacts_folder ~ radio['name'] }}.suit
+        - suit-directive-fetch:
+          - suit-send-record-failure
+        - suit-condition-image-match:
           - suit-send-record-success
           - suit-send-record-failure
           - suit-send-sysinfo-success
           - suit-send-sysinfo-failure
-      - - suit-condition-image-match:
+
+        - suit-directive-set-component-index: 0
+        - suit-directive-override-parameters:
+            suit-parameter-uri: '#{{ application['name'] }}'
+            suit-parameter-image-digest:
+              suit-digest-algorithm-id: cose-alg-sha-256
+              suit-digest-bytes:
+                envelope: {{ artifacts_folder ~ application['name'] }}.suit
+        - suit-directive-fetch:
+          - suit-send-record-failure
+        - suit-condition-image-match:
           - suit-send-record-success
           - suit-send-record-failure
           - suit-send-sysinfo-success
           - suit-send-sysinfo-failure
-    - suit-condition-dependency-integrity:
-      - suit-send-record-success
-      - suit-send-record-failure
-      - suit-send-sysinfo-success
-      - suit-send-sysinfo-failure
-    - suit-directive-process-dependency:
-      - suit-send-record-success
-      - suit-send-record-failure
-      - suit-send-sysinfo-success
-      - suit-send-sysinfo-failure
+{%- endif %}
+{%- if radio is defined %}
+      - - suit-directive-set-component-index: 0
+        - suit-directive-override-parameters:
+            suit-parameter-uri: '#{{ radio['name'] }}'
+            suit-parameter-image-digest:
+              suit-digest-algorithm-id: cose-alg-sha-256
+              suit-digest-bytes:
+                envelope: {{ artifacts_folder ~ radio['name'] }}.suit
+        - suit-directive-fetch:
+          - suit-send-record-failure
+        - suit-condition-image-match:
+          - suit-send-record-success
+          - suit-send-record-failure
+          - suit-send-sysinfo-success
+          - suit-send-sysinfo-failure
+        # Radio payload is available and correct. Proceed with radio verification.
+
+{%- if RAD_LOCAL_1_VERSION_MAJOR > 0 and application is defined %}
+        - suit-directive-set-component-index: {{ app_component_index }}
+        - suit-directive-override-parameters:
+            suit-parameter-version:
+              # Verify that the major version of the new radio manifest is greater from the
+              # currently installed application manifest major version by at most 1.
+              suit-condition-version-comparison-greater-equal: {{ RAD_LOCAL_1_VERSION_MAJOR - 1 }}.0.0
```

Comment:
I'm still worried about allowing to update major numbers in the "ladder" scheme...

Reply:
I do not understand what do you mean... I think this requires some offline discussion.